### DISCUSSION TITLE
OnDetails.HasDelegate

### DISCUSSION THREAD

**Discussion by 01Raja** [fretje/heroplate#15] [Q&A]
Do we have a working example with this feature? Can someone please help me with this? I can't find the complete code for this.

@if (OnDetails.HasDelegate)
{
      <HeroTooltipButton Tooltip="@L["Details"]" Icon="@Icons.Material.Filled.MoreHoriz"
             OnClick="() => OnDetails.InvokeAsync(context)" />
 }

**Comment by fretje** [comment]
You can use it like this:

```razor
<EntityTable TEntity="ProductDto" TId="int" TRequest="UpdateProductRequest" Context="@Context"
    OnDetails="@(product => Navigation.NavigateTo($"/products/{product.Id}"))">

    ...

</EntityTable>
```

In this case, it will simply navigate to the specified url, but you could also show a popup for instance.

**Comment by 01Raja** [comment]
Thanks.
